src/config_file.rs

```diff
@@ -144,27 +141,6 @@ impl Config {
         serde_yaml::from_str::<Self>(yaml)
     }
 
-    /// This provides the path for a configuration file, according to the XDG_BASE_DIRS specification.
-    /// return None if error like PermissionDenied
-    #[cfg(not(windows))]
-    pub fn config_file_path() -> Option<PathBuf> {
-        use xdg::BaseDirectories;
-        match BaseDirectories::with_prefix(CONF_DIR) {
```

Comment:
> with this change, can we handle the cases where people change the `XDG_CONFIG_HOME` env?

The `xdg::BseDirectories` was never available on Windows, since `xdg` doesn't support Windows at all. See [here](https://github.com/whitequark/rust-xdg/blob/f7e701e06ee9d29269b3edbd30f37494fcca2400/src/base_directories.rs#L300)

It is possible to just add more possible paths ourselves, now that this can support multiple paths

Reply:
And internally, `dirs` already uses this variable anyway, so we are using it too, see [here](https://docs.rs/dirs/5.0.1/dirs/fn.config_dir.html)

Reply:
this is used in not windows, according to https://docs.rs/dirs/5.0.1/dirs/fn.config_dir.html, linux will remain the XDG env, but not macOS, it seems like a breaking change for macOS users?

Reply:
@zwpaper hmm interesting, didn't think of mac :( 
Looks like internally `dirs` also [uses this] (https://docs.rs/dirs-sys/0.4.1/src/dirs_sys/lib.rs.html#115), but I can't test or verify that on my own because I don't have a mac. I'll review the code better to see if that can be corrected

Reply:
it breaks macos: https://github.com/dirs-dev/dirs-rs/blob/main/src/mac.rs#L10C54-L10C54

xdg crate would check the env in macOS

I tested locally

Reply:
Understood, I'll find a solution to solve this problem and revert back to xdg on any OS that isn't Windows